src/backends/torch/native/templates/tt/positionalenc.cc

```diff
@@ -0,0 +1,73 @@
+/**
+ * DeepDetect
+ * Copyright (c) 2019-2020 Jolibrain
+ * Author:  Guillaume Infantes <guillaume.infantes@jolibrain.com>
+ *
+ * This file is part of deepdetect.
+ *
+ * deepdetect is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * deepdetect is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with deepdetect.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "positionalenc.h"
+using namespace torch::indexing;
+
+namespace dd
+{
+  void PositionalEncodingImpl::init()
+  {
+
+    if (_dropout_ratio != 0.0)
+      _dropout
+          = register_module("dropout", torch::nn::Dropout(_dropout_ratio));
+
+    torch::Tensor pet;
+    if (_et == naive)
+      {
+        pet = torch::arange(0, _seq_len * _datadim,
+                            torch::TensorOptions(torch::kFloat32))
+                  .reshape({ _seq_len, _datadim });
+        pet = (pet / (float)(_seq_len * _datadim)) * 2.0 - 1.0;
+      }
+    else // sincos
+      {
```

Comment:
Do we want/can log the selected PE ? Also, maybe let's check on `sincos` and error if unknown ?

Reply:
ok for error.
PE type should be logged from ttransformer at construction time, it was easier than putting logger in every class.